GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java

```diff
@@ -58,178 +66,642 @@ public String GardenToString()
+              {
+                  Plant newPlant = new Plant(x, y, newGarden.speciesList.get(j));
+                  newGarden.plantList.add(newPlant);
+                  break;
+              }
+              
+              j++;                
+          }                       
+      }
+      
+      return newGarden;
+    }
+    
+    //done
+    public ArrayList<Plant> getPlantList()
+    {
+    	ArrayList<Plant> newList = new ArrayList<Plant>(plantList);
+        return newList;
+    }
+    
+    //done
+    public String[] getSpeciesNames()
+    {
+        int listSize = speciesList.size();
+        String speciesNames[] = new String[listSize];
         int i = 0;
-        int x, y;
         
-        //multiply by 5 and do i+= 5 because there are five items per species
-        for (i = 1; i < (numSpecies * 5); i+=5)
+        for (Species s : speciesList)
         {
-            type = parsedGarden[i];
-            sun = parsedGarden[i+1];
-            prune = Integer.parseInt(parsedGarden[i+2]);
-            water = Integer.parseInt(parsedGarden[i+3]);
-            fertilize = Integer.parseInt(parsedGarden[i+4]);
-            
-            Species newSpecies = new Species(type, sun, prune, water, fertilize);
-            speciesList.add(newSpecies);
+            String newString = new String(s.name);
+            speciesNames[i] = newString;
+            i++;
         }
         
-        
-        int numPlants = Integer.parseInt(parsedGarden[i]);
-        
-        i++;
-        
-        plantList = new ArrayList(numPlants);
-        
-        int j = 0;
-        int counter = numPlants * 3 + i;
-        
-        for (i = i; i < counter; i+= 3)
+        return speciesNames;
+    }
+    
+    //done
+    public Species getSpeciesInfo(String speciesName)
+    {
+    	Species newSpecies = null;
+        for (Species s : speciesList)
         {
-            x = Integer.parseInt(parsedGarden[i]);
-            y = Integer.parseInt(parsedGarden[i+1]);
-            type = parsedGarden[i+2];
-            j = 0;
-            
-            while (true)
+            if (s.name.equals(speciesName))
             {
-                if (speciesList.get(j).type.equalsIgnoreCase(type))
-                {
-                    Plant newPlant = new Plant(x, y, speciesList.get(j));
-                    plantList.add(newPlant);
-                    break;
-                }
-                
-                j++;                
-            }                       
+            	
+            	newSpecies = new Species(s.name, s.des, s.sun, s.type, (Date) s.plantDate.clone(), (Date) s.pruneDate.clone(), s.color, s.size);
+                break;
+            }
         }
         
-        return true;
-    }    
-	
-	/**
-	 * Returns a list of current species
-	 * <p>
-	 * 
-	 * @return String array of species
-	 */		
-    public String[] ListSpecies()
+        return newSpecies;
```

Comment:
188-201

This doesn't seem like an appropriate name for what the method returns